datalad/metadata/parsers/bids.py

```diff
@@ -8,59 +8,34 @@
 
-from os.path import exists, join as opj
 from datalad.support.json_py import load as jsonload
-from .. import _get_base_dataset_metadata
-
-# XXX Could become a class attribute
-_metadata_fname = 'dataset_description.json'
-
-
-# XXX Could become a class method
-def has_metadata(ds):
-    return exists(opj(ds.path, _metadata_fname))
-
-
-# XXX Could become a class method
-# XXX consider RFing into get_metadata(ds) and then centrally updating base_metadata
-#     with returned meta, or do we foresee some meta without that base?
-def get_metadata(ds, ds_identifier):
-    """Extract metadata from BIDS datasets.
-
-    Parameters
-    ----------
-    ds : dataset instance
-      Dataset to extract metadata from.
-
-    Returns
-    -------
-    dict
-      JSON-LD compliant
-    """
-    if not has_metadata(ds):
-        raise ValueError("no BIDS metadata found at {}".format(ds.path))
-
-    bids = jsonload(opj(ds.path, _metadata_fname))
-
-    meta = _get_base_dataset_metadata(ds_identifier)
-
-    # TODO maybe normalize labels of standard licenses to definition URIs
-    # perform mapping
-    for bidsterm, dataladterm in (('Name', 'name'),
-                                  ('License', 'license'),
-                                  ('Authors', 'author'),
-                                  ('ReferencesAndLinks', 'citation'),
-                                  ('Funding', 'foaf:fundedBy'),
-                                  ('Description', 'description')):
-        if bidsterm in bids:
-            meta[dataladterm] = bids[bidsterm]
-    compliance = ["http://docs.datalad.org/metadata.html#v0-1"]
-    # special case
-    if 'BIDSVersion' in bids:
-        compliance.append(
-            'http://bids.neuroimaging.io/bids_spec{}.pdf'.format(
-                bids['BIDSVersion'].strip()))
-    else:
-        compliance.append('http://bids.neuroimaging.io')
-    meta['dcterms:conformsTo'] = compliance
-    return meta
+from datalad.metadata.parsers.base import BaseMetadataParser
+
+
+class MetadataParser(BaseMetadataParser):
+    _core_metadata_filenames = ['dataset_description.json']
+
+    def _get_metadata(self, ds_identifier, meta, full):
+        bids = jsonload(
+            self.get_core_metadata_filenames()[0])
+
+        # TODO maybe normalize labels of standard licenses to definition URIs
+        # perform mapping
+        for bidsterm, dataladterm in (('Name', 'name'),
+                                      ('License', 'license'),
+                                      ('Authors', 'author'),
+                                      ('ReferencesAndLinks', 'citation'),
+                                      ('Funding', 'foaf:fundedBy'),
+                                      ('Description', 'description')):
+            if bidsterm in bids:
+                meta[dataladterm] = bids[bidsterm]
+        compliance = ["http://docs.datalad.org/metadata.html#v0-1"]
+        # special case
+        if 'BIDSVersion' in bids:
```

Comment:
I just merged a quick fix for this one, thus conflict -- "if bids.get('BIDSVersion')" since might be there but could be 'null' == None